### ISSUE TITLE
Cancelled jobs don't get cleaned up

### ISSUE THREAD

**Issue by Tomoka64** [sorentwo/oban#499]
### Precheck

Hi guys, I have been using Oban Job (scheduled jobs in particular. it is a very awesome library :)) 
One thing I faced is the "cancelled" jobs don't get cleaned up after the recycling happens. `Oban.cancel_job(job_id)` correctly sets the jobs to "cancelled", but they do not get removed like the other jobs with different states.
Am i missing any settings? I searched but could not find anything.

### Environment

- Oban Version: 2.6.1
- PostgreSQL Version: 10.15
- Elixir & Erlang/OTP Versions (`elixir --version`)
Erlang/OTP 23 [erts-11.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
Elixir 1.11.2 (compiled with Erlang/OTP 21)

### Current Behavior

After calling `Oban.cancel_job(job_id)` the state changes to `cancelled` and the job wont get executed but it remains in DB forever.

![Screen Shot 2021-06-08 at 4 00 34 PM](https://user-images.githubusercontent.com/32330406/121139587-de38f100-c873-11eb-9947-d0d457fed245.png)

### Expected Behavior

the cancelled jobs should be cleaned up like other jobs

**Comment by jaimeiniesta** [comment]
Can you share your Oban config? Are you using `Oban.Plugins.Pruner` or `Oban.Pro.Plugins.DynamicPruner`?

**Comment by Tomoka64** [comment]
Hi! thanks for responding. Here is my config
```
config :naboo, Oban,
  repo: XXX.Repo,
  plugins: [{Oban.Plugins.Pruner, max_age: 300}],
  queues: [default: 10, events: 50]
```
i am using `Oban.Plugins.Pruner`

**Comment by sorentwo** [comment]
@Tomoka64 Thanks for the report! Fixed with the linked commit 💛 

Pro's `DynamicPruner` handled this case, but it never came up for the basic pruner.

**Comment by Tomoka64** [comment]
are you planning to release soon? or do you recommend using DynamicPruner?

**Comment by sorentwo** [comment]
@Tomoka64 [Released with v2.72](https://hexdocs.pm/oban/changelog.html#2-7-2-2021-06-10)
